### PR TITLE
drivers: cc13xx_cc26xx: pwm: introduce pwm driver

### PR DESCRIPTION
**Pull request by ghost**
This change introduces a new PWM driver for all CC13/26xx SoC.

See the documentation in ti,cc13xx-cc26xx-timer-pwm.yaml for detailed usage instructions.

### CODE REVIEW

File: drivers/pwm/pwm_cc13xx_cc26xx_timer.c [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+#include <driverlib/gpio.h>
+#include <driverlib/prcm.h>
+#include <driverlib/timer.h>
+#include <inc/hw_memmap.h>
+#include <inc/hw_types.h>
+#include <ti/drivers/Power.h>
+#include <ti/drivers/power/PowerCC26XX.h>
+
+#include <zephyr/logging/log.h>
+#define LOG_MODULE_NAME pwm_timer
```
Comment (vaishnavachath) [inline drivers/pwm/pwm_cc13xx_cc26xx_timer.c:22]: @fgrandel , should this be something similar to the  driver name?

**Review by vaishnavachath — DISMISSED** [review]
Thanks for your contribution @fgrandel , looks good to me, I will test the changes later today.

**Review by vaishnavachath — DISMISSED** [review]

**Review by cfriedt — COMMENTED** [review]
LGTM, but can we also have this driver added to a build_all suite?

**Comment by fgrandel** [comment]
> LGTM, but can we also have this driver added to a build_all suite?

@cfriedt Thanks for your review. Fixed. :-)

**Review by cfriedt — APPROVED** [review]

**Review by vaishnavachath — APPROVED** [review]

**Pull request merged**